arangod/Cluster/DBServerAgencySync.cpp

```diff
@@ -73,21 +77,31 @@ Result DBServerAgencySync::getLocalCollections(VPackBuilder& collections) {
   }
   DatabaseFeature& dbfeature = _server.getFeature<DatabaseFeature>();
 
-  VPackObjectBuilder c(&collections);
-
-  dbfeature.enumerateDatabases([&](TRI_vocbase_t& vocbase) {
+  for (auto const& dbname : dirty) {
+    TRI_vocbase_t* tmp = dbfeature.lookupDatabase(dbname);
+    if (tmp == nullptr) {
+      continue;
+    }
+    TRI_vocbase_t& vocbase = *tmp;
     if (!vocbase.use()) {
```

Comment:
We are first doing lookup here, and then use. If worst comes to worse, then somebody deletes the database inbetween.

I think this could be simplified by using `dbFeature.useDatabase(dbname)` instead.
This will return a nullptr if the db does not exist or a valid pointer that already is marked used.